### PR TITLE
TabController - remove from incubator and remove support for children in TabBar

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
TabController - remove from incubator and remove support for children in TabBar
+ some minor fixes.

## Changelog
TabController - remove from incubator and remove support for children in TabBar

### CODE REVIEW

File: generatedTypes/components/badge/index.d.ts [hunk 1]
```diff
@@ -344,7 +344,7 @@ declare const _default: React.ComponentClass<ViewProps & TouchableOpacityProps &
     /**
      * the badge size (default, small)
      */
-    size: number;
+    size?: number | undefined;
```
Comment (Inbal-Tish) [inline generatedTypes/components/badge/index.d.ts:347]: Why do you need to specify undefined type for an optional prop?

File: src/components/tabController/TabBar.tsx [hunk 2]
```diff
@@ -144,47 +139,27 @@ const TabBar = (props: Props) => {
...
   useEffect(() => {
-    if (propsChildren) {
-      LogService.warn('uilib: Please pass the "items" prop to TabController.TabBar instead of children');
-    }
-
-    if ((propsItems || children.current) && !contextItems) {
+    if (propsItems && !contextItems) {
       _registerTabItems();
     }
   }, []);
```
Comment (Inbal-Tish) [inline src/components/tabController/TabBar.tsx:165]: Why no dependencies here, like 'items'? Won't it cause unnecessary updates?

File: generatedTypes/components/badge/index.d.ts (hunk 1 above)
Comment (M-i-k-e-l) [reply generatedTypes/components/badge/index.d.ts:347]: I don't, this is the automatically generated `.d.ts` file (when running `npm run build:dev`).

File: src/components/tabController/TabBar.tsx (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/tabController/TabBar.tsx:165]: On the contrary, no dependencies means no re-renders (this is like `componentDidMount`).

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**